### PR TITLE
feat: switch to basic auth, move structs, cleanup server-requests #36 #44

### LINKED ISSUE #36: Authorization
_Proposal:_
Use Basic Auth method instead of custom headers. `Authorization: Basic <<BASE64DIGEST>>` The `BASE64DIGEST` is in the form of `username:password`.

Example header:
`Authorization: Basic dGVzdHVzZXI6MTIzNA==`
`dGVzdHVzZXI6MTIzNA==` = `testuser:1234`

- [x] rename package to `auth`
- [x] Switch to new authorization method
- [x] edit `api.yml` (for all existing endpoints!)

Stakeholder: @Flippchen @MarionHinkel

### LINKED ISSUE #44: Cleanup server-requests.http

### PR DESCRIPTION
**Pull request by mfloto**
Resolves #36 
Resolves #44

### CODE REVIEW

File: auth/errors.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package auth
+
+import "errors"
+
+// Either the username or the password is wrong.
+var ErrWrongCredentials = errors.New("wrong credentials!")
```
Comment (yannickkirschen) [inline auth/errors.go:6]: Style guide: remove exclamation mark (on all errors)

File: auth/errors.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+var ErrNoCredentials = errors.New("no credentials supplied!")
+
+// The user does not have the required permissions/role.
+var ErrInsufficientPermissions = errors.New("insufficient permissions!")
+
+// The authorization header is invalid.
+var ErrInvalidAuthHeader = errors.New("invalid authorization header!")
+
+// The root user cannot be deleted.
+var ErrCannotDeleteRoot = errors.New("Cannot delete root user")
```
Comment (yannickkirschen) [inline auth/errors.go:18]: Style guide: lower case (also next error)

File: auth/user-authentication-middleware.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Author: Maximilian Floto
+package auth
+
+import (
+	"net/http"
+)
+
+// Takes as parameters the function serving the endpoint, the minimum role, an array of functions that are not subject to authentication
+func UserAuthMiddleware(f func(http.ResponseWriter, *http.Request), role Role, unrestrictedMethods []string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (yannickkirschen) [inline auth/user-authentication-middleware.go:10]: Is it just GH, or why is there so much indentation?

Comment (mfloto) [reply auth/user-authentication-middleware.go:10]: For me it's just one tab / four spaces

**Review by yannickkirschen — APPROVED** [review]

**Pull request merged**